container_test.go

```diff
@@ -174,6 +178,10 @@ func TestContainerStart(t *testing.T) {
 func TestContainerOutput(t *testing.T) {
 	t.Parallel()
 
+	if runtime.GOOS == "darwin" {
+		t.Skip("test with echo isn't supported on darwin")
```

Comment:
Why?

Reply:
Linux binary execution from public images (elf/amd64) on runu runtime is experimental and still has several limitations (only vfork, etc).  Thus the tests only use static-binary version of executable, and this image doesn't have various binaries (e.g., coreutils, etc).

So we did't test such stuffs in this PR.  I think we can revert it once it supports well.

Reply:
I found creating `client_darwin_test.go`  would reduce the patchsize while addressing darwin-specific behaviors of the tests.  Will try it.

Reply:
I have investigated what darwin containerd can test with `container_test.go`, and found that some of tests (11 out of 28 tests) should be skipped.


```
--- SKIP: TestWaitStoppedProcess (0.00s)
--- SKIP: TestContainerCloseIO (0.00s)
--- SKIP: TestWaitStoppedTask (0.00s)
--- PASS: TestNewClient (0.00s)
--- PASS: TestContainerUpdate (0.35s)
--- SKIP: TestKillContainerDeletedByRunc (0.00s)
--- PASS: TestShortRunningTaskPid (1.67s)
--- PASS: TestContainerExecNoBinaryExists (2.02s)
--- SKIP: TestContainerHook (0.00s)
--- PASS: TestContainerKill (2.79s)
--- PASS: TestContainerNoBinaryExists (3.51s)
--- PASS: TestDeleteRunningContainer (2.02s)
--- SKIP: TestDeleteContainerExecCreated (0.00s)
--- PASS: TestContainerExtensions (0.55s)
--- PASS: TestContainerLabels (0.55s)
--- SKIP: TestProcessForceDelete (0.00s)
--- SKIP: TestContainerHostname (0.00s)
--- PASS: TestContainerInfo (0.30s)
--- PASS: TestContainerExitedAtSet (1.40s)
--- SKIP: TestContainerLargeExecArgs (0.00s)
--- SKIP: TestContainerExec (0.00s)
--- PASS: TestNewContainer (0.55s)
--- SKIP: TestContainerStart (0.00s)
--- PASS: TestContainerPids (1.70s)
--- PASS: TestClientReconnect (0.00s)
--- PASS: TestContainerOutput (1.91s)
--- PASS: TestTaskForceDelete (1.57s)
--- PASS: TestShimSockLength (7.50s)
```


Main obstacle is the containerd-shim can't report the exit status due to different handling of reaper on darwin (i.e., we cannot use `wait4(2)` as processes aren't children without PR_SET_CHILD_SUBREAPER support).

Thus, I'm going to include limited tests in this PR but will update later once the shim (and runtimes) can report the status.